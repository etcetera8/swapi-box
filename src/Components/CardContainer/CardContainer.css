.CardContainer {
  margin: 0 auto;
  width: 80%;
  min-height: 500px;
  display: flex;
  background: rgba(253,253,255, 0.2);
  display: flex;
  flex-direction: row;
  margin-bottom: 100px;
}