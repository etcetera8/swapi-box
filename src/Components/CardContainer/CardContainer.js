import React from 'react';
import Card from '../Card/Card';
import PlanetCard from '../PlanetCard/PlanetCard';
import VehicleCard from '../VehicleCard/VehicleCard';
import './CardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = ({people, planets, vehicles, favorites, setFavorite, activeCategory}) => {

  const favorited = (favorited) => {
    const match = favorites.find(card => card.favorite === favorited)
      return match ? "favorited" : "";
  }

  const peopleCards = people.map((person, index) => 
    <Card 
      className={favorited(person.favorite)}
      setFavorite={setFavorite}
      person={person}
      key={index} />
  )

  const planetCards = planets.map((planet, index) =>
    <PlanetCard
      className={favorited(planet.favorite)}
      setFavorite={setFavorite}
      planet={planet}
      key={index}
    />
  )

  const vehicleCards = vehicles.map((vehicle, index) =>
   <VehicleCard 
      className={favorited(vehicle.favorite)}
      setFavorite={setFavorite}
      vehicle={vehicle}
      key={index}     
    />
  )

  const favoriteCards = favorites.map((card, index) => {
    let pick = '';
    if (card.category === 'people') {
      pick = <Card 
        className={favorited(card.favorite)}
        setFavorite={setFavorite}
        person={card}
        key={index} />
    } else if (card.category === 'vehicles') {
      pick = <VehicleCard
        className={favorited(card.favorite)}
        setFavorite={setFavorite}
        vehicle={card}
        key={index}    
      />
    }
    else if (card.category === "planets") {
      pick = <PlanetCard
        className={favorited(card.favorite)}
        setFavorite={setFavorite}
        planet={card}
        key={index}
      />
      }
      return pick;
    })

    return (
     <section className='CardContainer'> 
      { activeCategory === null &&
        <h2 className="message">Welcome! <br /> <br /> Choose a category to get started with exploring the Star Wars Universe</h2>
      } 
      {activeCategory === 'people' && 
        peopleCards
      }
      {activeCategory === 'planets' && 
        planetCards
      }
      {activeCategory === 'vehicles' && 
        vehicleCards
      }
      {activeCategory === 'favorites' && favoriteCards.length === 0 && 
          <h2 className="message">No favorites selected <br /> <br /> Find some and come back here to see them!</h2>
      }
      {activeCategory === 'favorites' && favoriteCards.length > 0 && 
        favoriteCards
      }
      </section>
    ) 
}

CardContainer.propTypes = {
  people: PropTypes.array.isRequired,
  planets: PropTypes.array.isRequired,
  vehicles: PropTypes.array.isRequired,
  favorites: PropTypes.array.isRequired,
  setFavorite: PropTypes.func.isRequired,
  activeCategory: PropTypes.string
};

export default CardContainer;