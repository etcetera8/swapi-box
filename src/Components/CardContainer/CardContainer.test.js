import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {
  let wrapper;
  const mockFunction = jest.fn();
  const mockArray = [{name: 'luke', homeworld:'naboo', species:'human', population: '20'}]
  beforeEach( () => {
    wrapper = shallow(<CardContainer 
                        activeCategory='people' 
                        setFavorite={mockFunction}
                        favorites={[]}
                        people={mockArray}
                        planets={[]}
                        vehicles={[]}
                      />)
  })
  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })
})