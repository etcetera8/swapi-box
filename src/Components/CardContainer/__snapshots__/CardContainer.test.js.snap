// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot test 1`] = `
<section
  className="CardContainer"
>
  <Card
    className=""
    homeworld="naboo"
    key="0"
    name="luke"
    population="20"
    setFavorite={[Function]}
    species="human"
  />
</section>
`;
